enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,222 @@
+	//
+	// +optional
+	// +kubebuilder:default={"0.0.0.0/0"}
+	AllowedSourceRanges []CIDR `json:"allowedSourceRanges,omitempty"`
+}
+
+// CIDR is an IP address range in CIDR notation (for example, "10.0.0.0/8"
+// or "fd00::/8").
+// +kubebuilder:validation:Pattern=`^(([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])/([0-9]|[12][0-9]|3[0-2])$`
+type CIDR string
+```
+
+### Workflow Description
+
+1. A cluster admin who wants to limit access to the load balancer to a list of IP address ranges
+of their choice sets the `AllowedSourceRanges` field. 
+2. Ingress Operator reconciles the LoadBalancer Service and sets the `.Spec.LoadBalancerSourceRanges`
+based on `AllowedSourceRanges`.
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+This proposal will modify the `IngressController` API by adding a new 
+variable called `AllowedSourceRanges` to the `LoadBalancerStrategy` 
+struct type and a new type `CIDR` that is a string in CIDR notation.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The value set for `AllowedSourceRanges` will be used to set `.Spec.LoadBalancerSourceRanges` 
+of the LoadBalancer Service in front of the router in cloud environments.
+
+If the customer directly sets `.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service,
+its value will be used to set `AllowedSourceRanges` if and only if `AllowedSourceRanges`
+was not initially set. Otherwise, `.Spec.LoadBalancerSourceRanges` is overwritten by `AllowedSourceRanges`.
+
+### Risks and Mitigations
+
+One risk of this proposal is that customers may have already set the
+`.Spec.LoadBalancerSourceRanges` field of the LoadBalancer Service. 
+This risk is mitigated by using this value to set `AllowedSourceRanges` 
+if and only if `AllowedSourceRanges` was not initially set.
+
+### Drawbacks
+
+This enhancement may bring additional complexity for the customers who are 
+used to modify directly the `Spec.LoadBalancerSourceRanges` of the LoadBalancer 
+Service for the same purposes. 
+
+## Design Details
+
+### Open Questions [optional]
+
+1. On Kubernetes API reference, the field 
+[`loadBalancerSourceRanges`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#service-v1-core) 
+has the warning "This field will be ignored if the cloud-provider does not support the feature.".
+However, there is no information about which platforms support this. Is there any platform
+OpenShift runs on which does not support this feature?
+
+### Test Plan
+
+The controller that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.  The operator has end-to-end
+tests; for this enhancement, a test is added that verifies the operator 
+correctly configures the load balancer service and updates the service 
+when the allowed source ranges are changed.
```

Comment:
Could you enumerate the test cases? I think it might be important to spell out specific test cases so we don't forget (specifically, the second test is of interest to me):
1. Test adding/deleting/changing AllowedSourceRanges to IC and observing loadBalancerSourceRanges on the loadbalancer
2. Test adding loadBalancerSourceRanges to loadBalancer and observing AllowedSourceRanges on the IngressController in which it was unset previously.

Those are the main two I can think of, if I understand the proposal correctly. Thoughts?

Reply:
Agreed. I adjusted the EP.